### ISSUE TITLE
Namespaces for Taxonomy ... rich namespaces to separate source and build origins

### ISSUE THREAD

**Issue by hartsock** [package-url/purl-spec#129]
In #128, I discussed custom pURL for archives downloaded from semi-structured repositories on the internet. 

Briefly:
```
https://commondatastorage.googleapis.com/chromium-boringssl-fips/boringssl-ae223d6138807a13006342edfeef32e813246b39.tar.xz
```

... to pURL-ify this URL I might try to write:
```
pkg:archive/commondatastorage.googleapis.com/chromium-boringssl-fips#boringssl@ae223d6138807a13006342edfeef32e813246b39?format=tar.xz
```

In [my own musings](https://medium.com/@hartsock/global-supply-chain-security-purl-and-namespace-emerging-conventions-ccb05bd2c7a) I noticed that the namespace is _not_ used in all repository types and might be useful for some specialized cases.

Let's say my company builds all its libraries from source code locally and repackages it. Some times with some specialized bug fixes or non-standard enhancements for our special Linux distribution.

Consider: [tcping-1.3.5-1.el7.rf](http://rpmfind.net/linux/RPM/dag/redhat/el7/x86_64/tcping-1.3.5-1.el7.rf.x86_64.html) 

This complex version number tells us it is given name: `tcping` version: `1.3.5` relsease `1.el7.rf` which is a custom compile of tcping for Red Hat Linux `el7` arch `x86_64`

Maybe this version has some special RedHat only bug fixes in it that aren't in other builds of the component?

If my company were to take `tcping-1.3.5-1.el7.rf.src.rpm` and re-compile this yet again to create a wholly new RPM from the source code ... what should this _new_ rpm be named?

A candidate might be:
```
pkg:rpm/tcping.libs.example.com#redhat/tcping@1.3.5-1.el7.rf/tcping@1.3.5-1example.com?arch=x86_Strange
```

... which is trying to communicate that this _is_ the source for `tcping@1.3.5-1.el7.rf` but recompiled and repackaged at `example.com` for the imaginary `x86_Strange` architecture  ... (maybe RedHat doesn't produce "strange" architecture builds) ... and this new binary version is the version _example.com_ is releasing in an RPM.

*tl;dr*
* In the `tcping` example source code comes from the same "provenance" (source code or src.rpm file) but results in _different_ binary rpm files. Can we capture that? Should we?
* What maps a package's output back to its source(s)? Is that out of scope for package URL as they only indicate a pointer to the individual binary package?
* is all this potentially solved by just adopting #128 and using "custom archives" to hold these kinds of relationships?

**Comment by johnmhoran** [comment]
Hi @hartsock .  Thank you for opening this issue.  I'm working on helping to bring our issues and PRs up-to-date and see that @iamwillbar and @pombredanne appear to have answered the questions you raised.  If that's the case, can you please close this issue?  And if not, please advise what remains to be addressed.  Thanks!
